### PR TITLE
WIP(filter): add filter functionality to team-member page

### PR DESCRIPTION
**Pull request by kasszz**

### CODE REVIEW

File: components/team-members-overview/team-members-overview.js [hunk 1]
```diff
@@ -1,19 +1,113 @@
...
+		}
+	}
+
+	render() {
+		const {team, peopleTab, vacanciesOverview, vacancies} = this.props;
+		const {activeRoles, activeLocations} = this.state;
+		return (
+		<div>
+			<Filter name={this.roles.name} data={this.roles.data} activeData={activeRoles} clickHandler={this.filterOnClickHandler} />
+			<Filter name={this.locations.name} data={this.locations.data} activeData={activeLocations} clickHandler={this.filterOnClickHandler} />
```
Comment (Tj3rk) [inline components/team-members-overview/team-members-overview.js]: clickHandler can maybe renamed to something like `onFilter` that is a bit more descriptive.
and maybe `data` -> `filters` and `activeData` -> `activeFilter`

File: components/team-members-overview/team-members-overview.js [hunk 2]
```diff
@@ -1,19 +1,113 @@
...
+			return;
+		}
+	}
+
+	render() {
+		const {team, peopleTab, vacanciesOverview, vacancies} = this.props;
+		const {activeRoles, activeLocations} = this.state;
+		return (
+		<div>
+			<Filter name={this.roles.name} data={this.roles.data} activeData={activeRoles} clickHandler={this.filterOnClickHandler} />
```
[the remark this answers is not in the document]
Comment (Tj3rk) [reply components/team-members-overview/team-members-overview.js]: maybe just hardcode `name` here? so you don't need locations object anymore. Same goes for the roles

File: components/team-members-overview/team-members-overview.js [hunk 3]
```diff
@@ -1,19 +1,113 @@
...
+			<p className="team-members-intro-text container">{peopleTab.introText}</p>
+			<ul className="team-members-overview container">
+				{ team.map((teamMember, index) => {
+					if (teamMember.hide) { return; }
+
+					if(!activeLocations.includes(teamMember.location.location)) {
+						return;
+					}
+
+					const activeRole = teamMember.newRoles.map(role => {
```
Comment (Tj3rk) [inline components/team-members-overview/team-members-overview.js]: I would put this in a separate function, something like `getActiveRole`.  Can be outside `TeamMembersOverview` class

File: components/team-members-overview/team-members-overview.js [hunk 4]
```diff
@@ -1,19 +1,113 @@
...
+	render() {
+		const {team, peopleTab, vacanciesOverview, vacancies} = this.props;
+		const {activeRoles, activeLocations} = this.state;
+		return (
+		<div>
+			<Filter name={this.roles.name} data={this.roles.data} activeData={activeRoles} clickHandler={this.filterOnClickHandler} />
+			<Filter name={this.locations.name} data={this.locations.data} activeData={activeLocations} clickHandler={this.filterOnClickHandler} />
+			<p className="team-members-intro-text container">{peopleTab.introText}</p>
+			<ul className="team-members-overview container">
+				{ team.map((teamMember, index) => {
```
Comment (Tj3rk) [inline components/team-members-overview/team-members-overview.js]: maybe make a const on top of the render function called `filteredTeam` with a function `filterTeam(team)` where you do all the filtering for `hide`, `location` and `role`
so: `filteredTeam = filterTeam(team)`

File: components/team-members-overview/team-members-overview.js [hunk 5]
```diff
@@ -1,19 +1,113 @@
...
+
+class TeamMembersOverview extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const {team} = props;
+
+		this.roles = {
+			name: 'Roles',
+			data: this.getRoles(team)
```
Comment (Tj3rk) [inline components/team-members-overview/team-members-overview.js]: I think data can be an object with `[{value: foo, isActive: false}, {value: bar, isActive: false}]`

**Comment by kasszz** [comment]
The feature is done, but not ready for production. The mobile design still needs to be implemented.

**Pull request merged**